src/Lecture3.hs

```diff
@@ -213,8 +268,22 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f acc (List1 x xs) = foldr f acc ([x] <> xs)
```

Comment:
That's correct 👍🏻 
And you can slightly improve the performance by applying `f` first instead of creating a new list:

```suggestion
  foldr f acc (List1 x xs) = f x (foldr f acc xs)
```